### PR TITLE
Internationalisation

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
This PR adds `react-i18next` to `datagateway-dataview` and `datagateway-download`, which allows for text in the app to be sourced from a JSON file and thus configured at will. It also helps with some things like pluralisation in `datagateway-download`.

Note that most of the changes aren't really visible, since if everything is working correctly then the strings should just load and not look any different to before. The only exception is that I didn't make DLS/ISIS specific sections to the translation file, so they share the same default words (e.g. DLS used to say "beamline" instead of "instrument")

## Testing instructions
- [x] Review code
- [x] Check Travis build
- [x] Review changes to test coverage

## Agile board tracking
part of #84 (still need to do it for `datagateway-search`)

### CODE REVIEW

File: packages/datagateway-dataview/src/pageContainer.component.test.tsx [hunk 1]
```diff
@@ -14,6 +14,21 @@ import { createLocation } from 'history';
 import { MemoryRouter } from 'react-router';
 
 import PageContainer from './pageContainer.component';
+// import { I18nextProvider } from 'react-i18next';
```
Comment (tswinb) [inline packages/datagateway-dataview/src/pageContainer.component.test.tsx:17]: Did you mean to keep these comments in?

File: packages/datagateway-dataview/src/breadcrumbs.component.tsx [hunk 2]
```diff
@@ -248,12 +251,9 @@ class PageBreadcrumbs extends React.Component<
         base: {
           entityName: baseEntityName,
 
-          // TODO: This display name requires internationalisation,
-          //       as currently it adds an 's' to the word disregarding
-          //       the current language the application is being served in.
-          displayName:
-            `${baseEntityName}`.charAt(0).toUpperCase() +
-            `${baseEntityName}s`.slice(1),
+          displayName: this.props.t(`breadcrumbs.${baseEntityName}`, {
```
Comment (tswinb) [inline packages/datagateway-dataview/src/breadcrumbs.component.tsx:254]: There's something going wrong with the breadcrumb label for facility cycles
image

File: packages/datagateway-download/src/downloadTab/downloadTab.component.tsx [hunk 3]
```diff
@@ -94,23 +96,39 @@ const DownloadTabs: React.FC = () => {
         textColor="primary"
         centered
       >
-        <Tab label="Cart" aria-label="Cart tab" {...a11yProps(0)} />
-        <Tab label="Downloads" aria-label="Downloads tab" {...a11yProps(1)} />
+        <Tab
+          label="Cart"
```
Comment (tswinb) [inline packages/datagateway-download/src/downloadTab/downloadTab.component.tsx:100]: Should the tab labels be also 'translated' as well as the aria-labels?

Comment (louise-davies) [reply packages/datagateway-download/src/downloadTab/downloadTab.component.tsx:100]: Totally missed this, yes they should be!

File: packages/datagateway-dataview/src/pageContainer.component.test.tsx (hunk 1 above)
Comment (louise-davies) [reply packages/datagateway-dataview/src/pageContainer.component.test.tsx:17]: oops

**Pull request merged**